src/Microsoft.AspNetCore.Blazor.Razor.Extensions/TagHelperDescriptorExtensions.cs

```diff
@@ -125,5 +126,27 @@ public static string GetEventArgsType(this TagHelperDescriptor tagHelper)
             tagHelper.Metadata.TryGetValue(BlazorMetadata.EventHandler.EventArgsType, out var result);
             return result;
         }
+
+        /// <summary>
+        /// Gets the set of component attributes that can accept child content (<c>RenderFragment</c> or <c>RenderFragment{T}</c>).
+        /// </summary>
+        /// <param name="tagHelper">The <see cref="TagHelperDescriptor"/>.</param>
+        /// <returns>The child content attributes</returns>
+        public static IEnumerable<BoundAttributeDescriptor> GetChildContentProperties(this TagHelperDescriptor tagHelper)
+        {
+            if (tagHelper == null)
+            {
+                throw new ArgumentNullException(nameof(tagHelper));
+            }
+
+            for (var i = 0; i < tagHelper.BoundAttributes.Count; i++)
+            {
+                var attribute = tagHelper.BoundAttributes[i];
+                if (attribute.IsChildContentProperty())
+                {
+                    yield return attribute;
+                }
+            }
```

Comment:
I'm mildly curious about the reason for preferring this kind of construction over a LINQ `.Where(...)` expression. Of course there's nothing wrong with the code as-is here - just interested in your thoughts on the subject.

Reply:
This is more debuggable 😆 

I wouldn't say I have a personal preference for one or the other, if anything I tend to avoid both iterators and linq for explicit loops and lists/arrays.